src/main/scala/com/fulcrumgenomics/util/NumericCounter.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.util
+
+import dagr.commons.CommonsDef._
+
+/**
+  * Super-simple class for counting occurrences of any [Numeric]. Will return
+  * zero for any item that has not been counted yet.  Implements some useful methods
+  * to compute statistics when the objects being counted are numeric types.
+  */
+class NumericCounter[T](implicit n: Numeric[T]) extends SimpleCounter[T] {
+  import n._
+
+  /** Returns the mean as a [Double]. */
+  def mean(): Double = {
+    var len = 0L
+    val sum = iterator.map { case (k, v) => len += v; k.toDouble() * v }.sum
+    sum / len
+  }
+
+  /** Returns the standard deviation as a [Double]. */
+  def standardDeviation(): Double = {
+    val m: Double = mean()
+    var len = 0L
+    val sum = iterator.map { case (k, v) => len += v; v * Math.pow(k.toDouble() - m, 2) }.sum
+    Math.sqrt(sum / (len - 1))
+  }
+
+  /** Returns the key with the greatest count (mode of the distribution). */
+  def mode(): T = iterator.maxBy(_._2)._1
+
+  /** Returns the median key.  If the number of values is even, consider using [medianAsDouble]. */
+  def median(): T = this.div(medianTimesTwo(), fromInt(2))
+
+  /** Returns the median key as a [Double].  This may be useful when the number of values is even and so the
+    * median is an average. */
+  def medianAsDouble(): Double = medianTimesTwo().toDouble() / 2.0
+
+  /** Returns the median times two, so that methods can implement the divide as desired. */
+  private def medianTimesTwo(): T = {
```

Comment:
Holy method batman!  This seems complicated!  How about we change SimpleCounter to a) take a type tag and b) if the type extends Ordered we use a `SortedMap` instead of a `Map`?  That's probably a nice change to have in general, and this this method would boil down to something like:

```scala
val len = size
val sum = if (len % 2 == 1) {
    val item iterator.drop(len/2).head
    Iterator(item, item).sum
}
else {
    iterator.drop((len/2) - 1).take(2).sum
}

sum / 2
```